src/perf_counters.h

```diff
@@ -66,7 +70,38 @@ class PerfCounterValues {
             sizeof(uint64_t) * (kPadding + nr_counters_)};
   }
 
-  static constexpr size_t kPadding = 1;
+  // This reading is complex and as the goal of this class is to
+  // abstract away the intrincacies of the reading process, this is
+  // a better place for it
+  size_t Read(const std::vector<int>& leaders) {
```

Comment:
any reason why the implementation is in the header?

Reply:
No particular reason. I was just trying to follow the flow of your code. I can definitely move there.